src/providers/AuthProvider.tsx

```diff
@@ -18,26 +19,31 @@ export const useAuthenticationData = () => useContext(AuthenticationContext);
 export const useTokenDispatch = () => useContext(TokenDispatchContext);
 
 const AuthProvider = ({ children }: { children: React.ReactNode }) => {
-  const localStorageKey = 'token';
   const navigate = useNavigate();
 
   const [authentication, dispatch] = useReducer<AuthenticationReducer>(
     authenticationReducer,
-      {
-        token: ""
-      }
+    {
+      token: '',
+    }
   );
 
   useEffect(() => {
     if (authentication.token) {
       axios.defaults.headers.common['Authorization'] =
         'Bearer ' + authentication.token;
       localStorage.setItem(localStorageKey, authentication.token);
+      console.log('Have token');
     } else if (localStorage.getItem(localStorageKey)) {
```

Comment:
I believe we shouldn't access the localStorage directly. The data from localStorage should be copied into the reducer's state once the page loads, and then you should only use the state.

You can add an action like `refreshFromStore` to the `authentiaction.Actions.ts`, that reads localStorage and set the token if it's defined. Then in the `App.tsx` you can call that action a the root level to make sure it runs on page refreshes.

The approach should be:
- If I want to login, run an action on reducer that obtains a token and saves it in reducer's state and in localstorage,
- If you want to logout, run an action on reducer that clears the state and localstorage,
- If you refresh the page, call an action on reducer that loads data from localStorage an save it in state, or do nothing if storage is empty,
- If you want to access the data (check if user is logged), then only access the state, never localStorage, as it's already in sync because of rules above.

This way you have a single point where localStorage is mutated.

Reply:
But hey, if this is too much, merge it as this. It's much better than it was before.

Please don't feel pressured by this. 😎

Reply:
yes, I like your suggestion a lot. I will try to implement it :)

Reply:
> But hey, if this is too much, merge it as this. It's much better than it was before.
> 
> Please don't feel pressured by this. 😎

I am not, nice to have someone to read the code. I was starting to feeling lonely ;) Thanks a lot for the comments :)